File: backend/src/main/java/com/ems/backend/service/DashboardService.java
package com.ems.backend.service;

import com.ems.backend.dto.*;

import java.util.List;

public interface DashboardService {
    Response getDashboardStats();
    Response getEmployeesPerDepartment();
    Response getLeaveStatistics();
    Response getMonthlySalaryExpenses();
    Response getRecentActivities();
    Response getNotifications();
}

File: frontend/src/services/dashboardService.js
import api from "./api";

const API = "http://localhost:8080/api/dashboard";

export const getDashboardStats = () => api.get(`${API}/stats`);

export const getEmployeesPerDepartment = () => api.get(`${API}/employees-per-department`);

export const getLeaveStatistics = () => api.get(`${API}/leave-statistics`);

export const getMonthlySalaryExpenses = () => api.get(`${API}/monthly-salary-expenses`);

export const getRecentActivities = () => api.get(`${API}/recent-activities`);

export const getNotifications = () => api.get(`${API}/notifications`);
